### PR TITLE
Update training compiler notebook for TF-2.6.3 and adding example to tune batch size

### PR DESCRIPTION
**Pull request by access2rohit**
Description of changes:

- Update Training Compiler Tensorflow 2.6.3 IOC notebook example
- Add Batch tuning example for Tensorflow
    Testing done:
    testing on personal sagemaker Natobook

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [x] I have linted my notebook(s) and code using `tox -e black-format,black-nb-format`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by access2rohit** [comment]
@navinsoni @BasilBeirouti can you please review ?

File: sagemaker-training-compiler/huggingface/tensorflow_tune_batch_size/finding_max_batch_size_for_model_training.ipynb [hunk 1]
```diff
@@ -0,0 +1,387 @@
+{
```
Comment (jkroll-aws) [inline sagemaker-training-compiler/huggingface/tensorflow_tune_batch_size/finding_max_batch_size_for_model_training.ipynb:1]: <p>To define the acronym, the first time you mention Hugging Face, say <code>Hugging Face (HF)</code>. ﻿</p><p> </p><p>Consistently use a space, <code>Hugging Face</code>  , which is the official spelling. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3275/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (jkroll-aws) [inline sagemaker-training-compiler/huggingface/tensorflow_tune_batch_size/finding_max_batch_size_for_model_training.ipynb:1]: <p>Add a note about the estimated runtime (probably several hours? how many hours?)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3275/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (jkroll-aws) [inline sagemaker-training-compiler/huggingface/tensorflow_tune_batch_size/finding_max_batch_size_for_model_training.ipynb:1]: <p>Add a little more explanation about how you're finding the max batch size and what it means, how to use it in the future.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3275/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (jkroll-aws) [inline sagemaker-training-compiler/huggingface/tensorflow_tune_batch_size/finding_max_batch_size_for_model_training.ipynb:1]: <p>Use the paper name for link text: <code>Accurate, Large Minibatch SGD: Training ImageNet in 1 Hour </code>	. Also why are you linking that particular paper? How does it relate to SM Training Compiler?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3275/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

**Comment by Lokiiiiii** [comment]
@jkroll-aws On a more general note,
would you rather 
(1) we created new notebooks for every framework release, deprecating notebooks associated with unsupported versions or
(2) modify existing notebooks to reflect the latest version ?

**Comment by jkroll-aws** [comment]
> @jkroll-aws On a more general note, would you rather (1) we created new notebooks for every framework release, deprecating notebooks associated with unsupported versions or (2) modify existing notebooks to reflect the latest version ?

I would favor updating the existing notebooks for each new release - then the commit history reflects what changes are needed to upgrade, and older versions are still available for reference in the history. Deprecating a notebook can be vague and messy because we don’t want to create broken links or duplicates if you want to delete or create a fresh slightly-altered copy of a notebook.

File: sagemaker-training-compiler/huggingface/tensorflow_tune_batch_size/finding_max_batch_size_for_model_training.ipynb (hunk 1 above)
Comment (Lokiiiiii) [reply sagemaker-training-compiler/huggingface/tensorflow_tune_batch_size/finding_max_batch_size_for_model_training.ipynb:1]: Fixed

Comment (Lokiiiiii) [reply sagemaker-training-compiler/huggingface/tensorflow_tune_batch_size/finding_max_batch_size_for_model_training.ipynb:1]: Added more explanation.

Comment (Lokiiiiii) [reply sagemaker-training-compiler/huggingface/tensorflow_tune_batch_size/finding_max_batch_size_for_model_training.ipynb:1]: Added more explanation and fixed link text

**Comment by zhaoqizqwang** [comment]
Closing this PR since it was created a long time ago and didn't pass all the checks. Feel free to open a new pull request instead.

**Pull request closed without merging**